src/Cli/Action/InstallAliyunWafAction.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Acme PHP project.
+ *
+ * (c) Titouan Galopin <galopintitouan@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace AcmePhp\Cli\Action;
+
+use AcmePhp\Ssl\CertificateResponse;
+use AlibabaCloud\Client\AlibabaCloud;
+use AlibabaCloud\WafOpenapi\WafOpenapi;
+use AlibabaCloud\Client\Exception\ClientException;
+use AlibabaCloud\Client\Exception\ServerException;
+
+/**
+ * Action to install certificate in an Aliyun Waf.
+ *
+ * @author Xiaohui Lam <xiaohui.lam@aliyun.com>
+ */
+class InstallAliyunWafAction extends AbstractAction
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle($config, CertificateResponse $response)
+    {
+        $issuerChain = [];
+        $issuerCertificate = $response->getCertificate()->getIssuerCertificate();
+        while (null !== $issuerCertificate) {
+            $issuerChain[] = $issuerCertificate->getPEM();
+            $issuerCertificate = $issuerCertificate->getIssuerCertificate();
+        }
+        $cert = implode("\n", $issuerChain);
+
+        $key = $response->getCertificateRequest()->getKeyPair()->getPrivateKey()->getPEM();
+
+        try {
+            AlibabaCloud::accessKeyClient($config['accessKeyId'], $config['accessKeySecret'])->regionId('cn-hangzhou')->asDefaultClient();
+            $response = WafOpenapi::v20161111()->upgradeInstance()
+                ->host('wafopenapi.cn-hangzhou.aliyuncs.com')
+                ->action('CreateCertAndKey')
+                ->setProtocol('https')
+                ->version('2018-01-17')
+                ->withCert($cert)
+                ->withKey($key)
+                ->withDomain($config['domain'])
+                ->withHttpsCertName($config['domain'])
+                ->withInstanceId($config['instanceId'])
+                ->request();
+        } catch (ServerException $e) {
+            throw $e;
```

Comment:
what's the purpose of catching then rethrowing?

Reply:
Oh... here is i'm **not sure** if
```bash
~/composer/vendor/bin/acme run ~/acmephp.yml run
```
Runs, and an exception occur, shoud the exception throw directly and exit 1,

Or try next cert install?

Reply:
What I mean is :
```
try {
  work();
} catch (Foo $e) { throw $e;}
} catch (Bar $e) { throw $e;}
```
is equivalent to 
```
work();
```

Reply:
> What I mean is :
> 
> ```
> try {
>   work();
> } catch (Foo $e) { throw $e;}
> } catch (Bar $e) { throw $e;}
> ```
> 
> is equivalent to
> 
> ```
> work();
> ```

Okay, I see.